test/e2e/provisioning/aws_e2e_test.go

```diff
@@ -0,0 +1,92 @@
+// +build e2e
+
+package provisioning
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+)
+
+const (
+	aws_manifest = "./testdata/machine-aws.yaml"
+)
+
+var awsScenarios = []scenarios{
```

Comment:
What I actually meant was not only moving the type but the testdefinitions as well, since they are equal for all cloud providers. We should then add the `[]scenearios` as arg to the `TestCloudProviderProvisioninE2"` funcs